v3/response.go

```diff
@@ -102,9 +102,7 @@ func (r *searchResponse) start(ctx context.Context, searchRequest *SearchRequest
 			case <-ctx.Done():
 				r.conn.Debug.Printf("%d: %s", msgCtx.id, ctx.Err().Error())
 				return
-			default:
-				r.conn.Debug.Printf("%d: waiting for response", msgCtx.id)
```

Comment:
I suggest you move the debug line above the select statement instead of erasing it completely:
```go
foundSearchSingleResultDone := false
for !foundSearchSingleResultDone {
	r.conn.Debug.Printf("%d: waiting for response", msgCtx.id)
	select {
	case <-ctx.Done():...
```